src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/TagHelperAttributeList.cs

```diff
@@ -77,21 +67,8 @@ public TagHelperAttributeList(IEnumerable<TagHelperAttribute> attributes)
         /// <returns>The first <see cref="TagHelperAttribute"/> with <see cref="TagHelperAttribute.Name"/> matching
         /// <paramref name="name"/>.
         /// </returns>
-        /// <remarks><paramref name="name"/> is compared case-insensitively. When setting,
-        /// <see cref="TagHelperAttribute"/>s <see cref="TagHelperAttribute.Name"/> must be <c>null</c> or
-        /// case-insensitively match the specified <paramref name="name"/>.</remarks>
-        /// <example>
-        /// <code>
-        /// var attributes = new TagHelperAttributeList();
-        ///
-        /// // Will "value" be converted to a TagHelperAttribute with a null Name
-        /// attributes["name"] = "value";
-        ///
-        /// // TagHelperAttribute.Name must match the specified name.
-        /// attributes["name"] = new TagHelperAttribute("name", "value");
-        /// </code>
-        /// </example>
-        public new TagHelperAttribute this[string name]
+        /// <remarks><paramref name="name"/> is compared case-insensitively.</remarks>
+        public new object this[string name]
```

Comment:
As w/ https://github.com/aspnet/Razor/pull/682#discussion-diff-51338876, don't change the API here either.